torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -332,6 +405,13 @@ ShardingUtil::GetShardIndicesForDevices(
       int offset = i;
       std::vector<at::indexing::TensorIndex> indices;
       for (int j = tile_shape.size() - 1; j >= 0; j--) {
+        if (sharding.replicate_on_last_tile_dim() &&
+            j == tile_shape.size() - 1) {
+          // the last tile assignment dimension is replicated, which implies
+          // that the consecutive `tile_shape[j]` devices hold the replicated.
+          offset /= tile_shape[j];
+          continue;
+        }
```

Comment:
High-level question - how does `tile_shape` change when the tensor is partially replicated?

Right now, this logic makes sense to me if partial replication adds a dimension to track replica group and `tile_shape[j] == 1` on the replicated dimensions, is that a correct understanding?

Reply:
That's correct.